src/pages/journal/components/test-slot/test-slot.ts

```diff
@@ -84,22 +83,31 @@ export class TestSlotComponent implements SlotComponent, OnInit {
   }
 
   canStartTest(): boolean {
-    if (!this.appConfig.getAppConfig().journal.allowTests) {
-      return false;
-    }
-
-    const startDate = new DateTime(this.slot.slotDetail.start);
-    if (startDate.daysDiff(this.dateTimeProvider.now()) < 0) {
-      return false;
-    }
+    const { testPermissionPeriods } = this.appConfig.getAppConfig().journal;
+    const { testCategory } = this.slot.booking.application;
+    const { start } = this.slot.slotDetail;
+    const startAsDate = new Date(start);
+    startAsDate.setHours(0, 0, 0, 0);
+    const periodsPermittingStart = testPermissionPeriods.filter((period) => {
```

Comment:
Not sure why we need this `filter`? Can we not just grab the `to` and `from` values as we know the test category we're interested in?

Reply:
With lodash - `const { to, from } =  _.find(testPermissionPeriods, period => period === testCategory);

Reply:
There can be multiple periods for a given category. e.g. an early adopter is given specific access (first JSON object) before the rollout to every examiner (second JSON object, `null` date):

```json
"testPermissionPeriods": [
  {
    "testCategory": "B",
    "from": "2019-08-01",
    "to": "2019-09-01"
  },
  {
    "testCategory": "B",
    "from": "2019-09-01",
    "to": null
  }
]
```